src/Gizmos/gizmo.ts

```diff
@@ -331,6 +332,23 @@ export class Gizmo implements IDisposable {
                 lmat.copyFrom(bone.getWorldMatrix());
             }
             bone.markAsDirty();
+        } else if (this._attachedNode.getClassName() === "SpotLight" || this._attachedNode.getClassName() === "PointLight" || this._attachedNode.getClassName() === "DirectionalLight") {

```

Comment:
is it possible to use feature-detection here? You need a shadow-light, you can try detecting if the attachedNode has a function that only a shadow light has (like `position` or `direction`)

Reply:
I've just seen the GetTypeId. Using it instead.